### ISSUE TITLE
refine reporting for training/finetunning

### ISSUE THREAD

**Issue by alinelena** [ACEsuit/mace#839]
at the moment evaluation of the model (RMSE tables train/valid/test) happen only at the end. 
I would suggest there is an option that you can optionally evaluate these every Nth epoch. 

another issue is evaluating model on pt_head for fine tuning can be time consuming and non useful in some scenarious, would be nice to be able to disable the pt part.

Defaults would not change but we will add these extra options.

**Comment by RokasEl** [comment]
Once merged, #836 will allow you to choose which heads to evaluate. For tracking training progress, you could just use the weights and biases integration. Instructions are [here](https://github.com/ACEsuit/mace?tab=readme-ov-file#weights-and-biases-for-experiment-tracking).

**Comment by alinelena** [comment]
@RokasEl yes I missed isaac's pr before i wrote this, commented on it.

I would prefer to avoid external services... imagine you run on a machine with no internet access...

**Comment by vue1999** [comment]
To track progress, with  #825 one will be able to visualize the loss and RMSE changes as you train and evaluate the model on the valid/train set after every specified Nth epoch. Would this be a good alternative for printing the tables during training?


![Image](https://github.com/user-attachments/assets/08742a75-f5b0-4c35-a94c-2bfe0782e900)

**Comment by alinelena** [comment]
almost, test is missing, also tables have some other info like relative error and so on.

**Comment by IsaacParker30** [comment]
@alinelena I have added ability to print the error tables every Nth epoch in my pull request #836. I think in #825 @vue1999 added the ability to plot any metric you want (including rel errors) by specifying the tags/keys as arguments.

**Comment by RokasEl** [comment]
> almost, test is missing, also tables have some other info like relative error and so on.

@alinelena, I don't think you should ever be looking at test data during training, that's the whole point of having a validation subset. While it is less of an issue in our field since the training and test data should be the same distribution, it is still bad ML practice.

On the other hand, I agree that having relative errors would be nice.

**Comment by alinelena** [comment]
@RokasEl yes and know on the bad practice... here is practical need. what I noticed sometime things seem to improve on the validation set (loss keeps going down) but not on the test set. so would be nice to be able to see how we did on test at epoch 50, 100 or 200...

**Comment by ilyes319** [comment]
I agree one should not regularly log the test set. If you improve on the valid but not on the test, it means you have a wrong validation set.
You can always test different checkpoint  by hand in case it makes sense to do it. But having an official option for it is a potential incentive for a bad practice.

**Comment by alinelena** [comment]
> If you improve on the valid but not on the test, it means you have a wrong validation set.
exactly. 

indeed if checkpoit models are saved you can test by hand... but at the moment we simple save every checkpoint or just last best...
